packages/protocol/contracts/governance/EpochRewards.sol

```diff
@@ -0,0 +1,399 @@
+   * @param targetVotingYieldInitial The initial relative target block reward for voters.
+   * @param targetVotingYieldMax The max relative target block reward for voters.
+   * @param targetVotingYieldAdjustmentFactor The target block reward adjustment factor for voters.
+   * @param rewardsMultiplierMax The max multiplier on target epoch rewards.
+   * @param rewardsMultiplierUnderspendAdjustmentFactor Adjusts the multiplier on target epoch
+   *   rewards when the protocol is running behind the target gold supply.
+   * @param rewardsMultiplierOverspendAdjustmentFactor Adjusts the multiplier on target epoch
+   *   rewards when the protocol is running ahead of the target gold supply.
+   * @param _targetVotingGoldFraction The percentage of floating gold voting to target.
+   * @param _targetValidatorEpochPayment The target validator epoch payment.
+   * @dev Should be called only once.
+   */
+  function initialize(
+    address registryAddress,
+    uint256 targetVotingYieldInitial,
+    uint256 targetVotingYieldMax,
+    uint256 targetVotingYieldAdjustmentFactor,
+    uint256 rewardsMultiplierMax,
+    uint256 rewardsMultiplierUnderspendAdjustmentFactor,
+    uint256 rewardsMultiplierOverspendAdjustmentFactor,
+    uint256 _targetVotingGoldFraction,
+    uint256 _targetValidatorEpochPayment
+  )
+    external
+    initializer
+  {
+    _transferOwnership(msg.sender);
+    setRegistry(registryAddress);
+    setTargetVotingYieldParameters(targetVotingYieldMax, targetVotingYieldAdjustmentFactor);
+    setRewardsMultiplierParameters(
+      rewardsMultiplierMax,
+      rewardsMultiplierUnderspendAdjustmentFactor,
+      rewardsMultiplierOverspendAdjustmentFactor
+    );
+    setTargetVotingGoldFraction(_targetVotingGoldFraction);
+    setTargetValidatorEpochPayment(_targetValidatorEpochPayment);
+    targetVotingYieldParams.target = FixidityLib.wrap(targetVotingYieldInitial);
+    startTime = now;
+  }
+
+  /**
+   * @notice Returns the target voting yield parameters.
+   * @return The target, max, and adjustment factor for target voting yield.
+   */
+  function getTargetVotingYieldParameters() external view returns (uint256, uint256, uint256) {
+    TargetVotingYieldParameters storage params = targetVotingYieldParams;
+    return (params.target.unwrap(), params.max.unwrap(), params.adjustmentFactor.unwrap());
+  }
+
+  /**
+   * @notice Returns the rewards multiplier parameters.
+   * @return The max multiplier and under/over spend adjustment factors.
+   */
+  function getRewardsMultiplierParameters() external view returns (uint256, uint256, uint256) {
+    RewardsMultiplierParameters storage params = rewardsMultiplierParams;
+    return (
+      params.max.unwrap(),
+      params.adjustmentFactors.underspend.unwrap(),
+      params.adjustmentFactors.overspend.unwrap()
+    );
+  }
+
+  /**
+   * @notice Sets the target voting gold fraction.
+   * @param value The percentage of floating gold voting to target.
```

Comment:
Nit: Capitalization of "gold" is inconsistent throughout the file. Here it's lowercase, in the comment to `struct RewardsMultiplierParameters ` it's uppercase. I'd vote for "Gold" since "Celo Dollars" is uppercase and that feels natural.